dot/peerset/handler.go

```diff
@@ -77,6 +77,10 @@ func (h *Handler) RemovePeer(setID int, peers ...peer.ID) {
 
 // ReportPeer reports ReputationChange according to the peer behaviour.
 func (h *Handler) ReportPeer(rep ReputationChange, peers ...peer.ID) {
+	for _, pid := range peers {
+		logger.Debugf("reporting reputation change of %d to peer %s, reason: %s", rep.Value, pid, rep.Reason)
+	}
+
```

Comment:
A tiny bit out of scope, but please change this method to accept a single peer ID. I checked its references, and it always get passed a single peer. `ReportPeer` doesn't match accepting multiple peers in its signature as well.
That way you can simplify that debug logging for loop you have as well.

Reply:
It's a feature not a bug. Accepting multiple peers is a common theme amongst all peerset handler funcs.